// ngx-search-filter/esm2022/lib/ngx-search-filter.module.mjs
import { NgModule } from '@angular/core';
import { NgxFilterPipe } from './ngx-filter.pipe';
import * as i0 from "@angular/core";
export class NgxSearchFilterModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.0", ngImport: i0, type: NgxSearchFilterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "18.0.0", ngImport: i0, type: NgxSearchFilterModule, declarations: [NgxFilterPipe], exports: [NgxFilterPipe] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "18.0.0", ngImport: i0, type: NgxSearchFilterModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.0", ngImport: i0, type: NgxSearchFilterModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgxFilterPipe
                    ],
                    imports: [],
                    exports: [
                        NgxFilterPipe
                    ]
                }]
        }] });
